moshi/src/test/java/com/squareup/moshi/MapJsonAdapterTest.java

```diff
@@ -86,6 +86,27 @@
     assertThat(jsonAdapter.fromJson(jsonReader)).isEqualTo(null);
   }
 
+  @Test public void covariantValue() throws Exception {
+    // Important for Kotlin maps, which are all Map<K, ? extends T>.
+    JsonAdapter<Map<String, Object>> jsonAdapter =
+        mapAdapter(String.class, Types.subtypeOf(Object.class));
+
+    Map<String, Object> map = new LinkedHashMap<>();
+    map.put("boolean", true);
+    map.put("float", 42.0);
+    map.put("String", "value");
+
+    String asJson = "{\"boolean\":true,\"float\":42.0,\"String\":\"value\"}";
+
+    Buffer buffer = new Buffer();
+    JsonWriter jsonWriter = JsonWriter.of(buffer);
+    jsonAdapter.toJson(jsonWriter, map);
+    assertThat(buffer.readUtf8()).isEqualTo(asJson);
```

Comment:
i'd remove these 4 lines, since they're not related to this test.

Reply:
I wanted to have this part of the test to ensure that serialization and deserialization worked with the adapter as-configured.